go-controller/pkg/network-attach-def-controller/network_attach_def_controller.go

```diff
@@ -68,9 +68,10 @@ type NetworkControllerManager interface {
 }
 
 type networkNADInfo struct {
-	nadNames  map[string]struct{}
-	nc        NetworkController
-	isStarted bool
+	nadNames      map[string]struct{}
+	nc            NetworkController
+	isStarted     bool
+	nadInDeletion string
```

Comment:
is it one controller per NAD? and so at max you can only have one entity in this value?

Reply:
no, multiple NADs could share the same controller if they share the same network name.

Reply:
I see thanks that makes sense! so each controller manages single network so 1 thread per network and then inside that multiple namespace-NADs might be connected to that same network so namespace scoped NADs are being processed across multiple threads! I see..